modules/optimization/src/executioners/AdjointSolve.C

```diff
@@ -95,6 +104,10 @@ AdjointSolve::solve()
 
   // Solve the adjoint system
   solver.adjoint_solve(matrix, solution, rhs, tol, maxits);
```

Comment:
Line 109 is a really smart way of doing this.  For my clarification, the adjoint solve in line 106 is using the scaled system matrix from the forward system on line 90.   Do you think it matters that our solution is from A^T=lambda. 
 We should try this with a nonsymmetric A.  Regular scaling of the system Ax=b would be A*D^-1*Dx=b. The adjoint system would be A^Tx=b so (A*D^-1*D)^Tx=b -> (D^T*D^-T*A^T)x=b. 
 So I wonder if D needs to be transposed for the adjoint solve.  Maybe this doesn't matter because D is for blocks and not columns of matrix.

Reply:
Do we have an example of a nonsymmetric A in the tests that I could "borrow"

Reply:
Also a diagonal matrix transposed is just the same diagonal matrix.

Reply:
I didn't think of that.  You're right.  I'll try to get a thermal mechanical example we can try.